### PR TITLE
Fix issues with API 201

### PR DESCRIPTION
**Pull request by ericzolf**
FIX: quoting variables where not correctly propagated when set from the command line.
FIX: extended variables were not correctly written if content of file hadn't changed and file was read-only.
FIX: some new functions were needed with higher security level and had to be sorted accordingly.

There were also a few testing issues that needed fixing:

- fix timetest to use function instead of API dependent variables
- when using the RepoShadow class multiple times, as object, it needs to be initiliazed
- fix sequence in restore skipping owners mapping initiation
- add stringify function __str__ to ExtendedAttributes object
- skip restore and cmdline tests with too many internals for new API
- fix test return codes for rdiff-backup-delete

## Self-Checklist

- [x] changes to the code have been reflected in the documentation
- [x] changes to the code have been covered by new/modified tests
- [x] commit contains a description of changes relevant to users prefixed by DOC:, FIX:, NEW: and/or CHG:

### CODE REVIEW

**Comment by ericzolf** [comment]
Ready for review. And once this PR is merged, together with the other ones, I'll release the last 2.2 version aka 2.2.6.

File: src/rdiff_backup/rpath.py [hunk 1]
```diff
@@ -1677,6 +1677,8 @@ def copy_attribs(rpin, rpout):
     if Globals.change_ownership:
         rpout.chown(*map_owners.map_rpath_owner(rpin))
     if Globals.eas_write:
+        if not rpin.issym():  # make sure EAs can be written
```
Comment (ikus060) [inline src/rdiff_backup/rpath.py:1680]: I don'T understand this change

Comment (ikus060) [reply src/rdiff_backup/rpath.py:1680]: Is it related to an issue ?

Comment (ikus060) [reply src/rdiff_backup/rpath.py:1680]: Ok, so it was not reported as a bug.

Thanks for the update !

Comment (ericzolf) [reply src/rdiff_backup/rpath.py:1680]: No, I changed the code to force API 201 and uncovered afew bugs while running all tests.

**Comment by ericzolf** [comment]
I guess, that means that I can merge?

**Pull request merged**